crates/test-programs/src/bin/sockets_0_3_tcp_sample_application.rs

```diff
@@ -30,23 +31,24 @@ async fn test_tcp_sample_application(family: IpAddressFamily, bind_address: IpSo
                     client.send(data_rx).await.unwrap();
                 },
                 async {
-                    data_tx.send(vec![]).await.unwrap();
-                    data_tx.send(first_message.into()).await.unwrap();
+                    let (result, _) = data_tx.write(vec![]).await;
+                    // TODO: this should be `Complete(0)`, not cancelled?
+                    assert_eq!(result, StreamResult::Cancelled);
```

Comment:
@dicej this is another one where we'll need to talk, I don't know entirely what's happening here but at the component-model level I'm not sure how to distinguish "successfully wrote 0 items" vs "successfully cancelled"